### PR TITLE
feat: case-insensitive match for pathname

### PR DESCRIPTION
**Pull request by Sayan751**
Related to https://github.com/WICG/urlpattern/issues/148.
This PR adds support for case-insensitive match for `pathname`. It adds a `caseSensitivePath` flag to the init object.

Example:

```ts
import { strictEqual } from 'assert';

const pattern = new URLPattern({ pathname: '/home', caseSensitivePath: false });

const url = new URL('https://example.com/home');
strictEqual(pattern.test(url.href), true, url.href);

url.pathname = '/HOME';
strictEqual(pattern.test(url.href), true, url.href);
```

### CODE REVIEW

**Comment by kenchris** [comment]
We are matching the spec so maybe file an issue on the spec as well?

**Comment by Sayan751** [comment]
> We are matching the spec so maybe file an issue on the spec as well?

Thanks! I am continuing the discussion already in the linked issue.

**Comment by wanderview** [comment]
Note, I think we should land WPT changes upstream in chromium or wpt before bringing them here.  Chromium will auto-sync to wpt and other browsers, but we don't have that setup for the polyfill repo.

And thanks to @Sayan751 for working on this!

**Review by kenchris — APPROVED** [review]

**Comment by kenchris** [comment]
This looks good to me, have the upstream WPT changed landed @wanderview ?

I am fine with landing this and making a new release.

**Comment by kenchris** [comment]
ping @wanderview

**Comment by wanderview** [comment]
No they have not.  Sorry, for the delay.

**Comment by kenchris** [comment]
@wanderview so you want me to wait with landing this until that has landed right?

**Comment by wanderview** [comment]
I think that would be best.  I'll try to get to the upstream work this week.

File: test/urlpatterntestdata.json [hunk 1]
```diff
@@ -834,6 +834,825 @@
...
+    "inputs": [{ "protocol": "https", "hostname": "example.com",
+                 "pathname": "/foo/bar", "search": "otherquery",
+                 "hash": "otherhash" }],
+    "exactly_empty_components": [ "username", "password", "port", "search",
+                                  "hash" ],
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar",
+                 "baseURL": "https://example.com?otherquery#otherhash" }],
```
Comment (wanderview) [inline test/urlpatterntestdata.json:956]: This case perhaps is missing an `ignoreCase:true` on pattern definition?

File: test/urlpatterntestdata.json [hunk 2]
```diff
@@ -834,6 +834,825 @@
...
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar", "baseURL": "https://example.com?query#hash"}, { "ignoreCase": true }],
+    "inputs": [ "https://example.com/fOo/baR?query#hash" ],
+    "exactly_empty_components": [ "username", "password", "port" ],
+    "expected_match": {
+      "hash": { "input": "hash", "groups": {} },
+      "hostname": { "input": "example.com", "groups": {} },
+      "pathname": { "input": "/fOo/baR", "groups": {} },
```
Comment (wanderview) [inline test/urlpatterntestdata.json:988]: Also, do any of these test cases check for case insensitive matching in the hash or search components?  Most the ones I've seen are just checking pathname.

**Comment by wanderview** [comment]
Note, I think I'm going to simplify the test changes in my upstream CL.  I don't think we need this many test cases for ignoreCase.

**Comment by wanderview** [comment]
The spec changes have landed and I have sent the intent to ship in chrome M107 here:

https://groups.google.com/a/chromium.org/g/blink-dev/c/KgAdo3kB1wc/m/UN70zkeNAwAJ

Note, I recommend using the WPT tests from here instead of the WPT tests currently in this PR:

https://github.com/web-platform-tests/wpt/commit/6d152e4018ca4f45ce93eea4985c516a518c3ad6

Unfortunately I'm not sure how to make this big of a change on someone else's PR.

**Comment by kenchris** [comment]
Hi @Sayan751 do you want to update your PR? Or do you want us to redo it and give you credit?

**Comment by Sayan751** [comment]
@kenchris I am sorry for this, but I have no time to update the PR. Please feel free to make the changes as you see fit.

**Comment by wanderview** [comment]
We could just land this and then do a fast follow-up to sync the upstream WPT json file.

**Comment by kenchris** [comment]
Sure. Let's do that.

**Comment by kenchris** [comment]
I will file an issue to track it

**Comment by kenchris** [comment]
OK filed https://github.com/kenchris/urlpattern-polyfill/issues/106

**Pull request merged**